packages/flutter/lib/src/widgets/sliver.dart

```diff
@@ -260,7 +414,18 @@ class SliverGrid extends SliverMultiBoxAdaptorWidget {
   }
 }
 
+/// A sliver that contains a multiple box children that each fill the viewport.
+///
+/// [SliverFill] places its children in a linear array along the main axis. Each
+/// child is sized to fill the viewport, both in the main and cross axis.
+///
+/// See also:
+///
+///  * [SliverFixedExtentList], which has a configurable [itemExtent].
+///  * [SliverList], which does not require its children to have the same
+///    extent in the main axis.
```

Comment:
Your uber patch is too big for my puny little brain to understand.  We need to break it down.  Probably the easiest thing is to pull out the obviously beneficial changes to other parts of the system.